integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/document_store.py

```diff
@@ -772,13 +710,40 @@ async def _fulltext_retrieval_async(
 
         return [self._mongo_doc_to_haystack_doc(doc) for doc in documents]
 
-    @staticmethod
-    def _mongo_doc_to_haystack_doc(mongo_doc: Dict[str, Any]) -> Document:
+    def _mongo_doc_to_haystack_doc(self, mongo_doc: Dict[str, Any]) -> Document:
         """
         Converts the dictionary coming out of MongoDB into a Haystack document
 
         :param mongo_doc: A dictionary representing a document as stored in MongoDB
         :returns: A Haystack Document object
         """
-        mongo_doc.pop("_id", None)
+        mongo_doc.pop("_id", None)  # MongoDB's internal id doesn't belong into a Haystack document, so we remove it.
+        if self.content_field != "content":
+            mongo_doc["content"] = mongo_doc.pop(self.content_field, None)
+        if self.embedding_field != "embedding":
+            mongo_doc["embedding"] = mongo_doc.pop(self.embedding_field, None)
         return Document.from_dict(mongo_doc)
+
+    def _haystack_doc_to_mongo_doc(self, haystack_doc: Document) -> Dict[str, Any]:
+        """
+        Parses a Haystack Document to a MongoDB document.
+
+        :param haystack_doc: the Haystack Document to convert
+        :returns: the MongoDB document in a dictionary representation
+        """
+        mongo_doc = haystack_doc.to_dict(flatten=False)
+        if self.content_field != "content":
+            mongo_doc[self.content_field] = mongo_doc.pop("content", None)
+        if self.embedding_field != "embedding":
+            mongo_doc[self.embedding_field] = mongo_doc.pop("embedding", None)
+        if "sparse_embedding" in mongo_doc:
+            sparse_embedding = mongo_doc.pop("sparse_embedding", None)
+            if sparse_embedding:
```

Comment:
Minor thing not strictly related to changes, but here maybe a:

```python
if sparse_embedding is not None
```

would be a bit clearer? Your choice ;)